docs/developer-guide/getting-started-with-persistence.md

```diff
@@ -75,173 +43,193 @@ following code snippet to your module's `pom.xml` under the `<dependencies>` sec
+    private Integer age;
 
-    public MyEntity()
+    public PetEntity()
     {
     }
 
-    public String getProperty()
+    @Override
+    public Integer getAge()
     {
-        return property;
+        return age;
     }
 
-    public void setProperty(String property)
+    @Override
+    public void setAge(Integer age)
+    {
+        this.age = age;
+    }
+
+}
+```
+
+## Creating a `EntityTraversalAdapter`
+
+As mentioned above, besides `neo4j-ogm` and `spring-data-neo4j`, we were forced to use custom CRUD implementations based on Gremlin. This has its advantages, as it allows us to optimize OGM entities and make them faster than what the common `neo4j-ogm` provides out of the box. The main thing of the Gremlin based CRUD is `EntityTraversalAdapter` which is a strategy for create/update/read/delete operations. The concrete `EntityTraversalAdapter` provides [Anonymous Traversals](http://tinkerpop.apache.org/docs/current/tutorials/gremlins-anatomy/) for each operation of the specific entity type. These traversals are used in Gremlin-based repositories to perform common CRUD operations:
+
+- `fold` : to construct entity instance based on vertex/edge and its properties
+- `unfold` : to extract entity properties into vertex/edge and its properties
+- `cascade` : to cascade other vertices/edges within delete if needed
+
+The `EntityTraversalAdapter` implementations can also use each other to support relations between entities, inheritance and cascade operations.
+
+Below is the code example of `EntityTraversalAdapter` implementation for `PetEntity`:
+
+```java
+package org.carlspring.strongbox.gremlin.adapters;
+
+import static org.carlspring.strongbox.gremlin.adapters.EntityTraversalUtils.extractObject;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.tinkerpop.gremlin.process.traversal.Traverser;
+import org.apache.tinkerpop.gremlin.structure.Element;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.carlspring.strongbox.domain.Pet;
+import org.carlspring.strongbox.domain.PetEntity;
+import org.carlspring.strongbox.gremlin.dsl.EntityTraversal;
+import org.carlspring.strongbox.gremlin.dsl.__;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PetAdapter extends VertexEntityTraversalAdapter<Pet>
+{
+
+    @Override
+    public Set<String> labels()
     {
-        this.property = property;
+        return Collections.singleton("Pet");
     }
 
     @Override
-    public boolean equals(Object o)
+    public EntityTraversal<Vertex, Pet> fold()
     {
-        if (this == o)
-        {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass())
-        {
-            return false;
-        }
-        
-        MyEntity myEntity = (MyEntity) o;
+        return __.<Vertex, Object>project("uuid", "age")
+                 .by(__.enrichPropertyValue("uuid"))
+                 .by(__.enrichPropertyValue("age"))
```

Comment:
Could you perhaps add some details about these oddly named classes/variables? (I know it was Gremlin, or Cypher, but it would probably help others get a better understanding of what this black magic spell is all about :) ). Thanks! :)

Reply:
fixed